pipelines/build-test-tool-template.yaml

```
@@ -17,7 +17,23 @@ steps:
     inputs:
       arguments: '-c $(BuildConfiguration)'
 
-  - task: DotNetCoreCLI@2
-    displayName: Run tests
+  - pwsh: |
+      dotnet test -c $(BuildConfiguration) --no-build -- --report-trx --results-directory $(Build.ArtifactStagingDirectory)/artifacts/TestResults --coverage --coverage-output $(Build.ArtifactStagingDirectory)/artifacts/Coverage/coverage.cobertura.xml --coverage-output-format cobertura --hangdump --hangdump-filename hangdump.dmp
+    displayName: Run unit tests
+
+  - task: PublishTestResults@2
+    displayName: Publish TRX Test Results
```

Windows only, please
```suggestion
    displayName: Publish TRX Test Results
    condition: eq(variables['Agent.OS'], 'Windows_NT')    
```

---

I'm curious, why would this be Windows-only?

---

Code coverage is reported by path, and we have different root paths on different environments, so we end up with a lot of duplicated data to sift through. Windows has the broadest test coverage and is the OS that the team uses for development, so that's where we get the biggest bang for the buck

---

That's probably worth having everywhere. If you need to export it as an artifact, though, you'll need to give use an environment-specific name for each artifact. Our templates don't allow artifacts to be updated after creation